app/models/miq_worker/container_common.rb

```diff
@@ -44,15 +44,18 @@ def default_image
     def resource_constraints
       mem_threshold = self.class.worker_settings[:memory_threshold]
       cpu_threshold = self.class.worker_settings[:cpu_threshold_percent]
+      mem_default   = self.class.worker_settings[:memory_minimum]
+      cpu_default   = self.class.worker_settings[:cpu_minimum_percent]
 
-      return {} if !Settings.server.worker_monitor.enforce_resource_constraints || (mem_threshold.nil? && cpu_threshold.nil?)
+      return {} unless Settings.server.worker_monitor.enforce_resource_constraints
+      return {} unless mem_threshold || cpu_threshold || mem_default || cpu_default
 
-      {:limits => {}}.tap do |h|
-        h[:limits][:memory] = "#{mem_threshold / 1.megabyte}Mi" if mem_threshold
-        if cpu_threshold
-          millicores = ((cpu_threshold / 100.0) * 1000).to_i
-          h[:limits][:cpu] = "#{millicores}m"
-        end
+      {}.tap do |h|
+        h.store_path(:limits, :memory, format_memory_threshold(mem_threshold)) if mem_threshold
```

Comment:
Do we need the `if`s?  Isn't there always a value due to the `worker_base/defaults`?

Reply:
I'm having to think about this one.
A user may want to explicitly set it to nil.
Or a user may even clear out the default since kubernetes already has a default for the pods and no there is no reason for us to reinvent this wheel.

I'm also assuming as time goes on we will transfer more and more of our settings and miq server in general to kubernetes.